source/common/filter/tcp_proxy.h

```diff
@@ -30,7 +30,11 @@ namespace Filter {
   COUNTER(downstream_cx_tx_bytes_total)                                                            \
   GAUGE  (downstream_cx_tx_bytes_buffered)                                                         \
   COUNTER(downstream_cx_total)                                                                     \
-  COUNTER(downstream_cx_no_route)
+  COUNTER(downstream_cx_no_route)                                                                  \
+  COUNTER(upstream_pause_reading)                                                                  \
```

Comment:
No problem at all with the questions.

I haven't thought through it completely, but I agree that explicitly tracking flow control pause/resume sounds useful and probably what we want. So renaming sounds good. We can always add connection level read disable/enable stats later on if we want. I guess that would be my vote (just rename of what you are doing now). But will defer to your judgement.